### ISSUE TITLE
Dialog Manager to accept Options

### ISSUE THREAD

**Issue by sbacha** [microsoft/botbuilder-dotnet#3985]
### Use this [query](https://github.com/Microsoft/botbuilder-dotnet/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen++label%3Aenhancement) to search for the most popular feature requests.

**Is your feature request related to a problem? Please describe.**
I am using dialogmanager to run declarative adaptive dialog from another custom dialog stack.
I would like to pass input from 1 dialog to the declarative adaptive dialog.
dialog.begindialog has options (input) unfortunately dialog manager does not expose a parameter to leverage the options param.

**Describe the solution you'd like**
Dialog Manager to accept Options as a property to be used when invoking the rootdialog.

**Describe alternatives you've considered**
DialogManager dm = new DialogManager(new BeginDialog()
                                   {
                                       Options = options,
                                       Dialog = rootDialog
                                     })

I considered this approach but the declarative dialog (root dialog ) in this example does not fire the recognizer on the first activity but waits for the second input to run recognition.

**Additional context**

[enhancement]

**Comment by vishwacsena** [comment]
@sbacha in your begin dialog, are you able to set `activityProcessed` to `false`? That should cause the callee adaptive dialog to re-run its recognizer and evaluate triggers on first turn.

**Comment by sbacha** [comment]
@vishwacsena yeah that fixed the problem with the alternative solution :) thank you.
but I still see value in exposing option param in the dailog manager, instead of this work around.

**Comment by vishwacsena** [comment]
@sbacha did the solution/ workaround provided by @tomlm  worked for you? Do you need anything else here?

**Comment by gabog** [comment]
Hi @sbacha, I am triaging the backlog for r10, can you ping me over teams and let me know if you still need this?

**Comment by gabog** [comment]
I talked with @sbacha offline and the workaround worked but it would be good to have an option so I am tagging this one as P1 and move in into the SDK repo to see if we can address this as part of R10
